src/Sarif/Core/Run.cs

```diff
@@ -217,5 +219,41 @@ public bool ShouldSerializeColumnKind()
         public bool ShouldSerializeLogicalLocations() { return this.LogicalLocations.HasAtLeastOneNonDefaultValue(LogicalLocation.ValueComparer); }
 
         public bool ShouldSerializeNewlineSequences() { return this.NewlineSequences.HasAtLeastOneNonNullValue(); }
+
+        internal static void ComputePolicies(IEnumerable<ToolComponent> policies)
+        {
+            // checking if we have have policies
+            if (policies == null || policies.Count() == 0)
+            {
+                return;
+            }
+
+            PoliciesCache.Clear();
+            foreach (ToolComponent policy in policies)
+            {
+                foreach (ReportingDescriptor rule in policy.Rules)
+                {
+                    PoliciesCache[rule.Id] = rule.DefaultConfiguration.Level;
+                }
+            }
+        }
+
+        public void ApplyPolicies()
```

Comment:
>ApplyPolicies [](start = 20, length = 13)

Let's add a doc comment to this. 'Applies the policies contained in this run, if any, to remap result failure levels. When multiple policies remap the same rule, the last policy in the policies collection has precedence.'